From 430b28893bd3809c043fc4f3eb88a8208d757f68 Mon Sep 17 00:00:00 2001
From: NelloKudo <marshnelloosu@gmail.com>
Date: Mon, 15 Sep 2025 17:07:35 +0200
Subject: [PATCH 3/7] winebus: Disable Hidraw by default, add log message.

Disabling Hidraw is needed for most controllers to work without issues: using it though, breaks
PlayStation glyphs for games with native dualsense support. Hence, notify the user about it
in logs, as PROTON_ENABLE_HIDRAW=1 fixes it.
---
 dlls/winebus.sys/main.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/dlls/winebus.sys/main.c b/dlls/winebus.sys/main.c
index 290b36a8b96..2cc999f4297 100644
--- a/dlls/winebus.sys/main.c
+++ b/dlls/winebus.sys/main.c
@@ -500,6 +500,10 @@ static DWORD check_bus_option(const WCHAR *option, DWORD default_value)
         if (info->Type == REG_DWORD) return *(DWORD *)info->Data;
     }
 
+    /* Hidraw is disabled by default: add logging message about it*/
+    if (!wcsicmp(option, L"DisableHidraw"))
+        FIXME("Hidraw is disabled by default; to enable it (ex. fix PlayStation glyphs not showing), add PROTON_ENABLE_HIDRAW=1\n");
+
     return default_value;
 }
 
@@ -1215,7 +1219,7 @@ static void bus_options_init(void)
 {
     options.disable_sdl = !check_bus_option(L"Enable SDL", 1);
     if (options.disable_sdl) TRACE("SDL devices disabled in registry\n");
-    options.disable_hidraw = check_bus_option(L"DisableHidraw", 0);
+    options.disable_hidraw = check_bus_option(L"DisableHidraw", 1);
     if (options.disable_hidraw) TRACE("IOHID and UDEV hidraw devices disabled in registry\n");
     options.disable_input = check_bus_option(L"DisableInput", 0);
     if (options.disable_input) TRACE("UDEV input devices disabled in registry\n");

-- 
2.54.0

